ssl/record/rec_layer_s3.c

```diff
@@ -1119,27 +1151,37 @@ int ssl3_write_pending(SSL *s, int type, const unsigned char *buf, size_t len,
             !(s->mode & SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER))
         || (s->rlayer.wpend_type != type)) {
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_SSL3_WRITE_PENDING,
-                 SSL_R_BAD_WRITE_RETRY);
+                SSL_R_BAD_WRITE_RETRY);
         return -1;
     }
 
     for (;;) {
         /* Loop until we find a buffer we haven't written out yet */
         if (SSL3_BUFFER_get_left(&wb[currbuf]) == 0
-            && currbuf < s->rlayer.numwpipes - 1) {
+                && currbuf < s->rlayer.numwpipes - 1) {
             currbuf++;
             continue;
         }
         clear_sys_error();
         if (s->wbio != NULL) {
             s->rwstate = SSL_WRITING;
+            if (BIO_get_offload_tx(s->wbio) && type !=
+                    SSL3_RT_APPLICATION_DATA) {
+                BIO_set_offload_tx_ctrl_msg(s->wbio, type);
+            }
             /* TODO(size_t): Convert this call */
             i = BIO_write(s->wbio, (char *)
-                          &(SSL3_BUFFER_get_buf(&wb[currbuf])
-                            [SSL3_BUFFER_get_offset(&wb[currbuf])]),
-                          (unsigned int)SSL3_BUFFER_get_left(&wb[currbuf]));
+                    &(SSL3_BUFFER_get_buf(&wb[currbuf])
+                        [SSL3_BUFFER_get_offset(&wb[currbuf])]),
+                    (unsigned int)SSL3_BUFFER_get_left(&wb[currbuf]));
             if (i >= 0)
                 tmpwrit = i;
+
+            /* To prevent coalescing of control and data messages,
```

Comment:
put the start-comment on a line by itself for multiline comments.

Reply:
Done.